Bullet-dodging game/Bullet-Dodging Game in Processing.md

```diff
@@ -0,0 +1,2191 @@
+# Bullet-Dodging Game in Processing
+
+Processing is an open-source graphical library that uses the Java language. Designed for the visual arts community, Processing can be used to create interactive graphics such as data visualizations, digital art, and video games. In this blog, we're going to build a simple bullet-dodging game from scratch in Processing. All of the built-in functions mentioned below can be found on the [Processing reference page](https://processing.org/reference/).
+
+
+
+## Table of contents:
+
+* [Part I: Getting Started](#part-i:-getting-started)
+* [Part II: Adding the Player and the Bullets](#part-ii:-adding-the-player-and-the-bullets)
+
+* [Part III: Adding a Life and Scoring System](#part-iii:-adding-a-life-and-scoring-system)
+
+* [Part IV: Adding Enemies and Shooting](#part-iv:-adding-enemies-and-shooting)
+
+* [Part V: Implementing More Subclasses](#part-v:-implementing-more-subclasses)
+
+* [Part VI: Adding Images](#part-vi:-adding-images)
+
+* [Part VII: Adding Obstacles](#part-vii:-adding-obstacles)
+
+* [Part VIII: Pausing and Restarting the Game](#part-viii:-pausing-and-restarting-the-game)
+
+
+
+## Part I: Getting Started
+
+The Processing Development Environment (PDE) can be downloaded from [here](https://processing.org/download/).
+
+After launching the editor, you will see a text editor, a compiler, and a display window. Programs are written in the text editor and started by pressing the Run button.
+
+The program runs by automatically calling two special functions: `setup()` and `draw()`. The `setup()` function is run once, when the program starts, so it's where we initialize the canvas and variables. After running the `setup()` function, the code inside `draw()` will run continuously from top to bottom until the program is stopped. Let's write down the two function names first.
+
+If we run the program by pressing the Run button on the upper left corner (or Ctrl + R), a small canvas with the default size will appear.
+
+![](images/editor1.png)
+
+First, let's use `size()` and `background()` and to set up the size and color of our canvas. The `size()` function takes two arguments--width and height, whereas the `background()` function can take either RGB or the gray parameter (0-255, where 0 means black and 255 means white) depending on the number of arguments. Since the canvas size will stay the same, we should put `size()` in the `setup()` function. On the other hand, since we will clear the contents of the window and redraw the screen each time `draw()` is called, `background()` should be put at the beginning of the `draw()` function.
+
+![](images/editor2.png)
+
+
+
+## Part II: Adding the Player and the Bullets
+
+Now that the canvas has been set up, let's add the player on the screen. For simplicity, let's temporarily represent the player using `circle()`.

```

Comment:
you have absolutely not set up the canvas at this point...

if you want this to be a follow-along blog, I would add a section above with screenshots for setting up and using the P3 system. Don't expect people to be able to get started with it on their own, and even then make sure they have some started code with the structure you described above in place

if you want this to be an informative blog and not follow-along, then you would need to remove the large amounts of explicit code samples below and add them to a resource dump or source file reference at the end

Reply:
I'm sorry for the image rendering issue, but I do have two screenshots showing how to up the canvas. I have also modified the previous section a little bit. Hopefully this will make the setup procedure clearer.

About the blog preference, the setup part isn't long enough for an individual blog, and I think setting up the environment isn't too complicated, and. Still, the entire blog seems to have too much information, so would it be a good idea to split it into two at somewhere else, say, after the third section?